default_crypto.go

```diff
@@ -32,48 +32,58 @@ func (c DefaultCrypto) GenerateDH() (DHPair, error) {
 	var pubKey [32]byte
 	curve25519.ScalarBaseMult(&pubKey, &privKey)
 	return dhPair{
-		privateKey: privKey,
-		publicKey:  pubKey,
+		privateKey: privKey[:],
+		publicKey:  pubKey[:],
 	}, nil
 }
 
 // DH returns the output from the Diffie-Hellman calculation between
 // the private key from the DH key pair dhPair and the DH public key dbPub.
-func (c DefaultCrypto) DH(dhPair DHPair, dhPub Key) Key {
+func (c DefaultCrypto) DH(dhPair DHPair, dhPub Key) (Key, error) {
 	var (
 		dhOut   [32]byte
-		privKey [32]byte = dhPair.PrivateKey()
-		pubKey  [32]byte = dhPub
+		privKey [32]byte
+		pubKey  [32]byte
 	)
+	copy(privKey[:], dhPair.PrivateKey()[:32])
```

Comment:
Not sure what's more dangerous: (1) copying less than 32 bytes or (2) panic. In this case, `[:32]` does not change anything because `privKey` is 32-bytes long anyway so no more data will be copied there. However, `PrivateKey` returns something shorter, this will panic.

https://play.golang.org/p/XC99saZV8Wa

Reply:
Not sure how much we care about this implementation, is currently not used (we provide our own), so I'll be leaning toward just having it as an example

Reply:
Ok, let's keep it then. At some point, we can refactor this to not have arrays at all if `DHPair` and `Key` do not require it.

Reply:
Since you can now return an error, would it make sense to check that the length of `dhPair.PrivateKey()` is 32, and return an error if not?